test/e2e/shared/aws.go

```diff
@@ -445,17 +448,23 @@ func deleteResourcesInCloudFormation(prov client.ConfigProvider, t *cfn_bootstra
 	iamSvc := iam.New(prov)
 	temp := *renderCustomCloudFormation(t)
 	for _, val := range temp.Resources {
+		By(fmt.Sprintf("deleting the following resource: %s", val.AWSCloudFormationType()))
 		tayp := val.AWSCloudFormationType()
 		if tayp == configservice.ResourceTypeAwsIamRole {
 			role := val.(*cfn_iam.Role)
+			By(fmt.Sprintf("cleanup for role with name '%s'", role.RoleName))
 			Eventually(func(gomega Gomega) bool {
 				_, err := iamSvc.DeleteRole(&iam.DeleteRoleInput{RoleName: aws.String(role.RoleName)})
 				return awserrors.IsNotFound(err) || err == nil
 			}, 5*time.Minute, 5*time.Second).Should(BeTrue())
 		}
 		if val.AWSCloudFormationType() == "AWS::IAM::InstanceProfile" {
 			profile := val.(*cfn_iam.InstanceProfile)
-			_, _ = iamSvc.DeleteInstanceProfile(&iam.DeleteInstanceProfileInput{InstanceProfileName: aws.String(profile.InstanceProfileName)})
+			By(fmt.Sprintf("cleanup for profile with name '%s'", profile.InstanceProfileName))
```

Comment:
I also added cleanup for these guys. Let's see if that helps.

Reply:
Since the thing it complains about existing is actually an `ManagedPolicy` and not an IAM Role.